tests/locale/keymap_or_locale.pm

```diff
@@ -0,0 +1,98 @@
+# SUSE's openQA tests
+#
+# Copyright © 2016-2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Keyboard layout test in console and gdm after boot
+# Maintainer: Martin Loviska <mloviska@suse.com>
+
+use base "opensusebasetest";
+use strict;
+use testapi;
+use utils;
+
+sub verify_default_keymap_textmode {
+    my ($test_string, $tag, $tty, $console) = @_;
+
+    if (defined($tty)) {
+        send_key($tty);
+    }
+
+    if (defined($console)) {
+        select_console($console);
+    }
+    type_string($test_string);
+    assert_screen($tag, 40);
+}
+
+# preparation for locale tests
+# usage should be discussed in future updates, not discussed
+# motivation: issue https://bugzilla.suse.com/show_bug.cgi?id=1074988
+
+sub list_locale_settings {
+    my $program = shift;
+
+    if (!check_var('DESKTOP', 'textmode') && get_var('DESKTOP')) {
+        x11_start_program($program);
+        script_run('cat /etc/X11/xorg.conf.d/00-keyboard.conf');
+    }
+    else {
+        clear_console;
+    }
+    script_run('localectl status');
+    script_run('locale');
+    script_run('cat /etc/vconsole.conf');
+}
+
+sub verify_default_keymap_x11 {
+    my ($test_string, $tag, $program) = @_;
+
+    x11_start_program($program);
+    type_string($test_string);
+    assert_screen($tag, 40);
+}
+
+sub run {
+    my $expected   = get_required_var('INSTALL_KEYBOARD_LAYOUT');
+    my $keystrokes = '-1';
+
+    if ($expected eq 'us') {
```

Comment:
Need to handle this condition in different way, if the parameter is not set is 'us', is default, to be align with work in keyboard layout switching in Installer.